src/main/java/org/jenkinsci/plugins/workflow/steps/EnvStep.java

```diff
@@ -50,22 +52,32 @@
                 throw new IllegalArgumentException(pair);
             }
         }
-        this.overrides = new ArrayList<String>(overrides);
+        this.overrides = new ArrayList<>(overrides);
     }
     
     public List<String> getOverrides() {
         return overrides;
     }
+
+    @Override public StepExecution start(StepContext context) throws Exception {
+        return new Execution(overrides, context);
+    }
     
-    public static class Execution extends AbstractStepExecutionImpl {
+    public static class Execution extends StepExecution {
```

Comment:
Likely to cause [JENKINS-40909](https://issues.jenkins-ci.org/browse/JENKINS-40909).